src/core/server/saved_objects/service/lib/repository.test.js

```diff
@@ -424,9 +424,17 @@ describe('SavedObjectsRepository', () => {
 
     const getMockBulkCreateResponse = (objects, namespace) => {
       return {
-        items: objects.map(({ type, id }) => ({
+        items: objects.map(({ type, id, attributes, references, migrationVersion }) => ({
           create: {
             _id: `${namespace ? `${namespace}:` : ''}${type}:${id}`,
+            _source: {
+              [type]: attributes,
+              type,
+              namespace,
+              references,
```

Comment:
This 3k lines js test file is scary. We should really at least migrate it to TS at some point (not in this PR though)

Reply:
yeah I think the test are almost as complex as the implementation, so it's sometimes hard to know if you can trust your tests :(

Reply:
😬 FWIW I did take a stab at converting it to TS, but ran into a lot of issues and gave up. Agree a separate PR would be worthwhile -- though at that point it may be a good idea to modularize the Repository itself and break it up into smaller files along with the tests.